tests/conftest.py

```diff
@@ -36,13 +46,113 @@
     TrackSegmentDatasetBuilderProvider,
 )
 
-pytest_plugins = ["nicegui.testing.user_plugin"]
+pytest_plugins = ["nicegui.testing.plugin"]
+
+
+# --- Acceptance test collection control ---
+# Ensure acceptance tests only run in the dedicated GitHub Actions workflow
+# without modifying workflow files. We detect the workflow via the
+# GITHUB_WORKFLOW environment variable which is automatically set by GitHub.
+# Alternatively, developers can force running acceptance tests locally by setting
+# RUN_ACCEPTANCE=1 in their environment.
+ACCEPTANCE_WORKFLOW_NAME = "Acceptance Test With Pytest"
+
+
+def _is_acceptance_context() -> bool:
+    return (
+        os.environ.get("GITHUB_WORKFLOW") == ACCEPTANCE_WORKFLOW_NAME
+        or os.environ.get("RUN_ACCEPTANCE") == "1"
+    )
+
+
+def pytest_ignore_collect(path: Path, config: pytest.Config) -> bool:  # type: ignore[override] # noqa
```

Comment:
For benchmark and regression tests, this is not necessary. I prefer to use the same mechanics for both. @randy-seng Lets discuss which way to go.